scalafix/rules/src/main/scala/migrate/ExplicitImplicitsRule.scala

```diff
@@ -98,16 +98,42 @@ class ExplicitImplicitsRule[G <: Global](g: G) {
 
   }
 
-  private def toAddImplicit(originalTree: Tree, args: List[String], isEtaExpansion: Boolean): Patch =
-    if (isEtaExpansion) Patch.addRight(originalTree, "(_)" + "(" + args.mkString(", ") + ")")
-    else Patch.addRight(originalTree, "(" + args.mkString(", ") + ")")
+  private def toAddImplicit(originalTree: Tree, args: List[String], isEtaExpansion: EtaExpansion): Patch =
+    isEtaExpansion match {
+      case EtaExpansion.WontHandle => Patch.empty
+      case EtaExpansion.False      => Patch.addRight(originalTree, "(" + args.mkString(", ") + ")")
+      case EtaExpansion.True(size) => {
+        val etaExpansionArgs = (1 to size).map(_ => "_")
+        Patch.addRight(originalTree, s"(${etaExpansionArgs.mkString(", ")})" + "(" + args.mkString(", ") + ")")
+      }
+    }
 
   // if eta expansion, we need to rewrite to (_)(implicits)
-  private def isEtaExpansion(fun: G#Tree, original: Tree): Boolean =
-    fun match {
-      case _: g.Name                                                                                    => false
-      case g.Apply(fun, List(arg)) if original.isInstanceOf[Name] || original.isInstanceOf[Term.Select] => true
-      case _                                                                                            => false
-    }
+  private def isEtaExpansion(fun: G#Tree, original: Tree): EtaExpansion = {
+    // case where function values.map(provide _)
+    // this would need a rewrite values.map((provide _)(_)(implicit)) that we don't handle
+    val isTermEta = original.parent.forall(p => p.isInstanceOf[Term.Eta])
```

Comment:
> // this would need a rewrite values.map((provide _)(_)(implicit)) that we don't handle

I think that most of the time it would need to be rewritten to `value.map((provide _)(implicits)` in which case we probably want to handle

Reply:
ok it's doable.